### PR TITLE
Pol-171 Google delete unattached volumes

### PR DESCRIPTION
**Pull request by rshade**
### Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/google/unattached_volumes/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+This policy requires the Google Credential.  When applying the policy select the appropriate credentials
+from the list for your tenant.  If such credential doesn't exist please contact your cloud admin to create the Credential.
+
+The credential must contain the value *Google* in the Provider field.
+Refer to our documentation for more details on the [Credential Service](https://docs.rightscale.com/policies/users/guides/credential_management.html)
+
+### Google Required Permissions
+
+The following Google permissions must be allowed for the policy to run.
```
Comment (cdwilhelm) [inline cost/google/unattached_volumes/README.md]: review the README_GUIDELINES for the proper wording.

File: cost/google/unattached_volumes/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+- The `compute.disks.delete` permission
+- The `compute.disks.list` permission
+- The `compute.disks.get` permission
+- The `compute.regions.list` permission
+- The `resourcemanager.projects.get` permission
+
+Snapshot Create Permissions:
```
Comment (cdwilhelm) [inline cost/google/unattached_volumes/README.md]: no need to separate, keep them all in line.

File: cost/google/unattached_volumes/google_delete_unattached_volumes.pt [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+)
+
+###############################################################################
+# Permissions
+###############################################################################
+
+permission "perm_read_creds" do
+  actions   "rs_cm.show_sensitive","rs_cm.index_sensitive"
+  resources "rs_cm.credentials"
+end
```
Comment (cdwilhelm) [inline cost/google/unattached_volumes/google_delete_unattached_volumes.pt]: no longer needed, must be removed.

File: cost/google/unattached_volumes/google_delete_unattached_volumes.pt [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+  grant type: "jwt_bearer" do
+    iss cred("GCE_PLUGIN_ACCOUNT")
+    aud "https://www.googleapis.com/oauth2/v4/token"
+    additional_claims do {
+      "scope" => "https://www.googleapis.com/auth/compute",
+      "scope" => "https://www.googleapis.com/auth/cloud-platform"
+    } end
+    signing_key cred("GCE_PLUGIN_PRIVATE_KEY")
+  end
+end
```
Comment (cdwilhelm) [inline cost/google/unattached_volumes/google_delete_unattached_volumes.pt]: should be replaced with credentials

File: cost/google/unattached_volumes/google_delete_unattached_volumes.pt [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+    auth $auth_google
+    host "cloudresourcemanager.googleapis.com"
+    verb "GET"
+    path "/v1/projects"
+    pagination $google_pagination
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response,"projects[*]") do
+      field "project_name", jmes_path(col_item, "projectId")
```
Comment (cdwilhelm) [inline cost/google/unattached_volumes/google_delete_unattached_volumes.pt]: field name should match the extracted field.

File: cost/google/unattached_volumes/google_delete_unattached_volumes.pt [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+  code <<-EOS
+  res = []
+  _.each(ds_projects, function(project){
+    _.each(ds_zones, function(zone){
+      var result_json = {"project": project.project_name,"zone": zone.zone}
+      res.push(result_json)
+    })
+  })
+  EOS
+end
```
Comment (cdwilhelm) [inline cost/google/unattached_volumes/google_delete_unattached_volumes.pt]: I'm not sure this is really needed.  please evaluate and remove if it's not necessary.

File: cost/google/unattached_volumes/google_delete_unattached_volumes.pt [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+# Policy
+###############################################################################
+
+policy 'policy_google_unattached_volumes' do
+  validate_each $ds_unattached_volumes do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Google unattached volumes Discovered"
+    detail_template <<-EOS
+# Unattached volumes from the Google Cloud 
+# Google Old volumes
+| Project ID | Name | Size | Labels | Age In Days| Status |
```
Comment (cdwilhelm) [inline cost/google/unattached_volumes/google_delete_unattached_volumes.pt]: zone should be added and sorted.  place zone after Project Id

File: cost/google/unattached_volumes/google_delete_unattached_volumes.pt [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+end
+
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy 'policy_google_unattached_volumes' do
+  validate_each $ds_unattached_volumes do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Google unattached volumes Discovered"
```
Comment (cdwilhelm) [inline cost/google/unattached_volumes/google_delete_unattached_volumes.pt]: change case for Discovered.  should be discovered.

File: cost/google/unattached_volumes/google_delete_unattached_volumes.pt [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+###############################################################################
+
+escalation "esc_email" do
+  email $param_email
+end
+
+escalation "esc_approve_delete_volumes" do
+  request_approval  do
+    label "Approve Resource Deletion"
+    description "Approve escalation to run RightScale Cloud Workflow to delete Old volumes"
```
Comment (cdwilhelm) [inline cost/google/unattached_volumes/google_delete_unattached_volumes.pt]: should be unattached instead of old.

**Comment by rshade** [comment]
https://governance.rightscale.com/org/6/projects/7954/applied-policies/5e54407be4ca220001b6ca30
https://governance.rightscale.com/org/6/projects/7954/applied-policies/5e544099e4ca220001b6ca31

**Pull request merged**